packages/dx-react-grid/docs/guides/tree-data.md

```diff
@@ -0,0 +1,74 @@
+# React Grid - Tree Data
+
+The Grid component can show tree data. Use the corresponding plugins to manage the expanded rows state and show tree data programmatically or via the UI (special column with ability to expand row).
+
+## Related Plugins
+
+The following plugins implement tree data visualization:
+
+- [TreeDataState](../reference/tree-data-state.md) - controls the expanded rows state
+- [CustomTreeData](../reference/custom-tree-data.md) - converts custom formatted tree data to a supported format
+- [TableTreeColumn](../reference/table-tree-column.md) - renders the table column with row indent, toggle button and selection controls
+
+Note that [plugin order](./plugin-overview.md#plugin-order) is important.
+
+## Basic Setup
+
+Use the `TreeDataState`, `CustomTreeData` and `TableTreeColumn` plugins to set up a Grid with simple tree data structure.
+
+### Custom Grouping
+
+If the data has a hierarchical structure (already grouped), use the `CustomTreeData` plugin.
```

Comment:
> If the data has a hierarchical structure

It looks like this phrase is unnecessary. We use the `CustomTreeData` plugin for both data structures: hierarchical and plain.